tests/conftest.py

```diff
@@ -233,6 +238,24 @@ def name_map_composite(request, model_vision, layer_map_composite):
     return NameMapComposite(name_map=name_map)
 
 
+@pytest.fixture(scope='session')
+def mixed_composite(request, model_vision, name_map_composite, special_first_layer_map_composite):
+    '''Fixture to create NameLayerMapComposites based on an explicit
+    NameMapComposite and SpecialFirstLayerMapComposites.'''
+    composites = [name_map_composite, special_first_layer_map_composite]
+    return MixedComposite(composites)
```

Comment:
This could blow up a little bit, as the `name_map_composite` is also generated from the layer-maps, if I remember correctly (i.e. the number of tests is the number of layer maps squared).
It may also be worth to check if the `layer_map` is actually triggered at all, as the `name_map` may already cover all layers. Ideally the `layer_map` should have entries that are not covered by the `name_map`.

Reply:
Ah I see, but now I'm not sure what solution would be best to work around this.
Do you have an idea?

Reply:
And you're right, the part of the test for the `LayerMapComposite` is never run at all

Reply:
What do you think about adding a new fixture named `partial_name_map_composite`.
This would work similar to `name_map_composite` but would skip the first half of the layers from `layer_map_composite`

Reply:
I think that's a good idea. Maybe we can create a fixture that randomly samples half of the mapping:
```python
import random

@pytest.fixture(scope='session')
def pyrng(rng):
    return random.Random(rng.initial_seed())

@pytest.fixture(scope='session')
def partial_name_map_composite(name_map_composite, pyrng):
    '''Fixture to create a randomly sampled partial NameMapComposites.'''
    name_map = name_map_composite.name_map
    return NameMapComposite(name_map=pyrng.sample(name_map, len(name_map) // 2))
```
(Maybe it would also be better to have a seed fixture so we do not need to extract it from the torch rng state)

Reply:
I will fix this in a following PR, let's try to get this one merged first.